lib/tasks/db.rake

```diff
@@ -115,4 +115,109 @@ namespace :db do
       end
     end
   end
+
+  namespace :reload do
+    desc 'Reload the development database with a fresh copy from the production dump without downtime'
+    task development: :environment do
+      if EnvConfig.WCA_LIVE_SITE?
+        abort "This action is disabled for the production server!"
+      end
+
+      Dir.mktmpdir do |dir|
+        FileUtils.cd dir do
+          dev_db_dump_url = DbDumpHelper.public_s3_path(DbDumpHelper::DEVELOPER_EXPORT_SQL_PERMALINK)
+          local_file = "./dump.zip"
+
+          LogTask.log_task("Downloading #{dev_db_dump_url}") do
+            system("curl -o #{local_file} #{dev_db_dump_url}") || raise("Error while running `curl`")
+          end
+
+          LogTask.log_task("Unzipping dump.zip") do
+            system("unzip #{local_file}") || raise("Error while running `unzip`")
+          end
+
+          config = ActiveRecord::Base.connection_db_config.configuration_hash
+          database_name = config[:database]
+          temp_db_name = "#{database_name}_temp"
+
+          LogTask.log_task "Creating and loading temporary database '#{temp_db_name}'" do
+            # Create the temporary database
+            ActiveRecord::Base.establish_connection(config.merge(database: nil))
+            ActiveRecord::Base.connection.execute("DROP DATABASE IF EXISTS #{temp_db_name} ")
+            ActiveRecord::Base.connection.execute("CREATE DATABASE #{temp_db_name}")
+
+            # Disable certain checks for faster loading
+            DatabaseDumper.mysql("SET unique_checks=0", temp_db_name)
+            DatabaseDumper.mysql("SET foreign_key_checks=0", temp_db_name)
+            DatabaseDumper.mysql("SET autocommit=0", temp_db_name)
+
+            if Rails.env.development?
+              DatabaseDumper.mysql("SET GLOBAL innodb_flush_log_at_trx_commit=0", temp_db_name)
+            end
+
+            # Load the dump into the temporary database
+            DatabaseDumper.mysql("SOURCE #{DbDumpHelper::DEVELOPER_EXPORT_SQL}", temp_db_name)
+
+            # Commit any pending transactions
+            DatabaseDumper.mysql("COMMIT", temp_db_name)
+          end
+
+          # RENAME Database has been removed, that's why we need to swap tables
+          LogTask.log_task "Swapping tables between databases" do
+            ActiveRecord::Base.connection.execute("CREATE DATABASE #{database_name}_old")
+            # Re-establish the connection to the old database so we can swap
+            ActiveRecord::Base.establish_connection(config)
+            # Get the list of tables from the current database using ActiveRecord
+            current_tables = ActiveRecord::Base.connection.execute("SHOW TABLES").map { |row| row[0] }
```

Comment:
There is `ActiveRecord::Base.connection.tables` which does the exact same thing for the currently connected DB ;)

Reply:
ahhh ofc